components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -978,7 +944,8 @@ void ConversationHandler::MaybeUnlinkAssociatedContent() {
   // Only unlink if panel is closed and there is no conversation history.
   // When panel is open or has existing conversation, do not change the state.
   if (chat_history_.empty() && !IsAnyClientConnected()) {
-    SetShouldSendPageContents(false);
+    associated_content_manager()->AddContent(nullptr, /*notify_updated=*/false,
+                                             /*detach_existing_content=*/true);
```

Comment:
I find it a bit confusing to detach content via calling AddContent with null delegate, as the method document said "Adds a content delegate to the list of content delegates." and it's what its name suggested too.
Could we add a public DetachContent which takes notify_updated and call private DetachContentInternal (current DetachContent) and optionally notify the update?

Reply:
Also, why do we choose to use notify_updated=false here? What are the use cases we would call DetchContent without triggering `ConversationHandler::OnAssociatedContent`? I am just worried about unintentional changes might happen here as we were calling `MaybeSeedOrClearSuggestions()` and `MaybeFetchOrClearContentStagedConversation()` in `SetShouldSendPageContents` always when should_send changes before.

Reply:
Added a |ClearContents| method which notifies observers - Don't remember why I set notify to false here 😆